app/components/RecoveryCodeList/index.js

```diff
@@ -2,27 +2,32 @@
 
 import React from "react";
 import { createFragmentContainer, graphql } from 'react-relay/compat';
+import type { RecoveryCodeList_recoveryCodes } from './__generated__/RecoveryCodeList_recoveryCodes.graphql';
 
 type Props = {
-  recoveryCodes: {
-    codes: Array<string>
-  }
+  recoveryCodes: RecoveryCodeList_recoveryCodes
 };
 
 class RecoveryCodeList extends React.PureComponent<Props> {
   render() {
+    if (!this.props.recoveryCodes.codes) {
+      return null;
+    }
+
     return (
-      <ul
-        className="list-reset center"
-        style={{
-          columns: 2
-        }}
-      >
-        {this.props.recoveryCodes.codes.map((code) => (
-          <li key={code}>
-            <code className="monospace h2">{code}</code>
-          </li>
-        ))}
+      <ul className="list-reset center" style={{ columns: 2 }}>
+        {
+          this.props.recoveryCodes.codes.map(({ code, consumed }, index) => (
+            <li key={index}>
+              <code
+                className="monospace h2"
+                style={{ textDecoration: (consumed ? 'line-through' : 'none') }}
```

Comment:
TODO: This is pretty janky — probably should be a proper component to allow us to have the hover behaviour and to tidy up the styling.